pandas/core/reshape/reshape.py

```diff
@@ -634,20 +652,7 @@ def _convert_level_number(level_num, columns):
         level_to_sort = _convert_level_number(0, this.columns)
         this = this.sort_index(level=level_to_sort, axis=1)
 
-    # tuple list excluding level for grouping columns
-    if len(frame.columns.levels) > 2:
```

Comment:
does frame.columns vs this.columns matter here?  we use both in the existing code but you're only passing one

Reply:
The only context in which `frame.columns` is used is `len(frame.columns.levels)` and the previous block of code (lines 635--646) do not change the number of levels (it is just swapping levels). So it's a bit simpler to use the rolled-version of the columns (`this.columns`) everywhere.